src/Casts/EnumCollectionCast.php

```diff
@@ -50,7 +50,7 @@ public function get($model, string $key, $value, array $attributes)
      */
     public function set($model, string $key, $value, array $attributes)
     {
-        if (is_null($value)) {
+        if (is_null($value) || empty($value)) {
```

Comment:
Hey,
I was thinking a lot on that one and it's something specific to the `SET/COMMA` format. The default `JSON` format perfectly handles this case.
The issue of the `COMMA` format is that the `explode` call creates an array with an empty string as one value. So it's more a deserialization issue.
The check should be in the `fromDatabase()` method and return an empty array if it's an empty string.
That one here manipulates the value - an empty array is another value than `null`. And that's what the JSON format maintains, so both formats should behave the same here. Empty array in and empty array out.

Reply:
@Gummibeer 

Sorry for being late with my changes. I've just pushed the requested changes.